src/contracts/account/ProxyAccountDeployer.sol

```diff
@@ -22,63 +36,83 @@ contract ProxyAccount is ReplayProtection {
-            // IF we slice onchain, then we lose that information.
-            emit Revert(string(revertReason));
-        }
+        // Assumes that ProxyAccountDeployer is ReplayProtection. 
+        bytes memory encodedData = abi.encode(CallType.DELEGATE, _metaTx.target, _metaTx.value, _metaTx.callData);
+
+        // // Reverts if fails.
+        require(owner == verify(encodedData, _replayProtection, _replayProtectionAuthority, _signature), "Owner did not sign this meta-transaction.");
+        delegate(_metaTx.target, _metaTx.callData); // We do not want this to revert. Save the replay protection 
     }
 
     /**
-     * User deploys a contract in a deterministic manner.
-     * It re-uses the replay protection to authorise deployment as part of the salt.
-     * @param _initCode Initialisation code for contract
-     * @param _replayProtectionAuthority Identify the Replay protection, default is address(0)
+     * A batch of meta-transactions or meta-deployments.
+     * One replay-protection check covers all transactions. 
+     * @param _metaTxList List of revertable meta-transactions
+     * @param _replayProtection Replay protection
+     * @param _replayProtectionAuthority Address of external replay protection
      * @param _signature Signature from signer
      */
-    function deployContract(
-        bytes memory _initCode,
+    function batch(RevertableMetaTx[] memory _metaTxList,
         bytes memory _replayProtection,
         address _replayProtectionAuthority,
         bytes memory _signature) public {
 
-        // Confirm the user wants to deploy the smart contract
-        require(owner == verify(_initCode, _replayProtection, _replayProtectionAuthority, _signature), "Owner of proxy account must authorise deploying contract");
+        bytes memory encodedData = abi.encode(CallType.BATCH, _metaTxList);
 
-        // We can just abuse the replay protection as the salt :)
-        address deployed = Create2.deploy(keccak256(abi.encode(owner, _replayProtection)), _initCode);
+        // Reverts if fails.
+        require(owner == verify(encodedData, _replayProtection, _replayProtectionAuthority, _signature), "Owner did not sign this meta-transaction.");
+
+        // Go through each revertable meta transaction and/or meta-deployment.
+        for(uint i=0; i<_metaTxList.length; i++) {
+
+            bool success = false; 
 
-        emit Deployed(owner, deployed);
+            if(_metaTxList[i].callType == CallType.CALL) {
+                success = forwardCall(_metaTxList[i].target, _metaTxList[i].value, _metaTxList[i].callData);
+            } else if(_metaTxList[i].callType == CallType.DELEGATE) {
+                success = delegate(_metaTxList[i].target, _metaTxList[i].callData);
+            }
+
+            // Should we fail on revert?
+            if(_metaTxList[i].revertOnFail) {
+                require(success, "Transaction reverted.");  
+            }
+        }
     }
 
     /**
```

Comment:
Cant the proxyAccountDeployer be removed now? We could use a global deployer to deploy the base account, and the batch function on the relayHub to deploy new clones.

Reply:
id be great not to mix up the relayhub/proxy account purposes. We can make a new ticket to deal with deployment properly though.

Reply:
Mmm, ok, can you create a ticket to remove the proxyAccountDeployer then.

Reply:
done